### DISCUSSION TITLE
Best value for slidingWindowSize

### DISCUSSION THREAD

**Discussion by kczecior** [resilience4j/resilience4j#1845] [Q&A]
Hello all,

I am new to CB, and I am trying to set the best value for slidingWindowSize. I know default size is 100 but isn't it better if we have a bigger slidingWindowSize.  Below are the properties I want to use, please comment.

slidingWindowSize = 500
minimumNumberOfCalls = 100
slidingWindowType = COUNT_BASED
automaticTransition = false
failureRateThresholdPercent = 20
permittedNumberOfCallsInHalfOpenState = 10
waitDurationInOpenStateInSeconds = 30

**Comment by RobWin** [comment]
Hi,

I can't give you any recommendations. It really depends on your workload (throughput) of your system.

**Reply by kczecior** [reply]
Hi RobWin,

Thank you for your answer, apologies for the delayed answer but I had to gather some data.
The max amount of TPS is 330 and the average TPS is 35

**Reply by RobWin** [reply]
It might be even simpler to use a TIME_BASED sliding window type and slidingWindowSize 10 in order to calculate the failure rate based on the calls of the last 10 seconds.

**Reply by kczecior** [reply]
Thank you very much for your suggestion.
